### ISSUE TITLE
Unable to run nccl tests for Torch 2.1.0 (nightly) with NCCL 2.12.12

### ISSUE THREAD

**Issue by ajindal1** [pytorch/pytorch#99669]
### 🐛 Describe the bug

NCCL all gather is failing for torch 2.1.0 (nightly) with NCCL 2.12.12. Code and error are provided below:

```python
import torch
import torch.cuda.nccl as nccl
import torch.cuda

nGPUs = torch.cuda.device_count()

cpu_inputs = [torch.zeros(128).uniform_().to(dtype=torch.float) for i in range(nGPUs)]
expected = torch.cat(cpu_inputs, 0)

inputs = [cpu_inputs[i].cuda(i) for i in range(nGPUs)]
outputs = [torch.zeros(128 * nGPUs, device=i, dtype=torch.float)
            for i in range(nGPUs)]
nccl.all_gather(inputs, outputs)

```

Error logs with NCCL_DEBUG=INFO:
```bash
>>> nccl.all_gather(inputs, outputs)
3ccc2fa29ce8:192:192 [0] NCCL INFO Bootstrap : Using eth0:172.17.0.2<0>
3ccc2fa29ce8:192:192 [0] NCCL INFO Plugin Path : /opt/nccl-rdma-sharp-plugins/lib/libnccl-net.so
3ccc2fa29ce8:192:192 [0] NCCL INFO P2P plugin IBext
3ccc2fa29ce8:192:192 [0] NCCL INFO NET/IB : No device found.
3ccc2fa29ce8:192:192 [0] NCCL INFO NET/IB : No device found.
3ccc2fa29ce8:192:192 [0] NCCL INFO NET/Socket : Using [0]eth0:172.17.0.2<0>
3ccc2fa29ce8:192:192 [0] NCCL INFO Using network Socket
NCCL version 2.12.12.MSCCL.0.7.3+cuda11.7
3ccc2fa29ce8:192:224 [0] NCCL INFO Setting affinity for GPU 0 to fff000
3ccc2fa29ce8:192:225 [1] NCCL INFO Setting affinity for GPU 1 to fff000
3ccc2fa29ce8:192:227 [3] NCCL INFO Setting affinity for GPU 3 to 0fff
3ccc2fa29ce8:192:226 [2] NCCL INFO Setting affinity for GPU 2 to 0fff
3ccc2fa29ce8:192:224 [0] NCCL INFO Channel 00/02 :    0   1   2   3
3ccc2fa29ce8:192:224 [0] NCCL INFO Channel 01/02 :    0   1   2   3
3ccc2fa29ce8:192:225 [1] NCCL INFO Trees [0] 2/-1/-1->1->0 [1] 2/-1/-1->1->0
3ccc2fa29ce8:192:227 [3] NCCL INFO Trees [0] -1/-1/-1->3->2 [1] -1/-1/-1->3->2
3ccc2fa29ce8:192:224 [0] NCCL INFO Trees [0] 1/-1/-1->0->-1 [1] 1/-1/-1->0->-1
3ccc2fa29ce8:192:226 [2] NCCL INFO Trees [0] 3/-1/-1->2->1 [1] 3/-1/-1->2->1
3ccc2fa29ce8:192:225 [1] NCCL INFO Channel 00 : 1[200000] -> 2[300000] via direct shared memory
3ccc2fa29ce8:192:224 [0] NCCL INFO Channel 00 : 0[100000] -> 1[200000] via direct shared memory
3ccc2fa29ce8:192:227 [3] NCCL INFO Channel 00 : 3[400000] -> 0[100000] via direct shared memory
3ccc2fa29ce8:192:225 [1] NCCL INFO Channel 01 : 1[200000] -> 2[300000] via direct shared memory
3ccc2fa29ce8:192:226 [2] NCCL INFO Channel 00 : 2[300000] -> 3[400000] via direct shared memory
3ccc2fa29ce8:192:224 [0] NCCL INFO Channel 01 : 0[100000] -> 1[200000] via direct shared memory
3ccc2fa29ce8:192:227 [3] NCCL INFO Channel 01 : 3[400000] -> 0[100000] via direct shared memory
3ccc2fa29ce8:192:226 [2] NCCL INFO Channel 01 : 2[300000] -> 3[400000] via direct shared memory
3ccc2fa29ce8:192:224 [0] NCCL INFO Connected all rings
3ccc2fa29ce8:192:227 [3] NCCL INFO Connected all rings
3ccc2fa29ce8:192:225 [1] NCCL INFO Connected all rings
3ccc2fa29ce8:192:226 [2] NCCL INFO Connected all rings
3ccc2fa29ce8:192:227 [3] NCCL INFO Channel 00 : 3[400000] -> 2[300000] via direct shared memory
3ccc2fa29ce8:192:227 [3] NCCL INFO Channel 01 : 3[400000] -> 2[300000] via direct shared memory
3ccc2fa29ce8:192:225 [1] NCCL INFO Channel 00 : 1[200000] -> 0[100000] via direct shared memory
3ccc2fa29ce8:192:226 [2] NCCL INFO Channel 00 : 2[300000] -> 1[200000] via direct shared memory
3ccc2fa29ce8:192:225 [1] NCCL INFO Channel 01 : 1[200000] -> 0[100000] via direct shared memory
3ccc2fa29ce8:192:226 [2] NCCL INFO Channel 01 : 2[300000] -> 1[200000] via direct shared memory
3ccc2fa29ce8:192:224 [0] NCCL INFO Connected all trees
3ccc2fa29ce8:192:224 [0] NCCL INFO threadThresholds 8/8/64 | 32/8/64 | 8/8/64 | 8/8/512
3ccc2fa29ce8:192:224 [0] NCCL INFO 2 coll channels, 2 p2p channels, 2 p2p channels per peer
3ccc2fa29ce8:192:227 [3] NCCL INFO Connected all trees
3ccc2fa29ce8:192:227 [3] NCCL INFO threadThresholds 8/8/64 | 32/8/64 | 8/8/64 | 8/8/512
3ccc2fa29ce8:192:227 [3] NCCL INFO 2 coll channels, 2 p2p channels, 2 p2p channels per peer
3ccc2fa29ce8:192:225 [1] NCCL INFO Connected all trees
3ccc2fa29ce8:192:225 [1] NCCL INFO threadThresholds 8/8/64 | 32/8/64 | 8/8/64 | 8/8/512
3ccc2fa29ce8:192:225 [1] NCCL INFO 2 coll channels, 2 p2p channels, 2 p2p channels per peer
3ccc2fa29ce8:192:226 [2] NCCL INFO Connected all trees
3ccc2fa29ce8:192:226 [2] NCCL INFO threadThresholds 8/8/64 | 32/8/64 | 8/8/64 | 8/8/512
3ccc2fa29ce8:192:226 [2] NCCL INFO 2 coll channels, 2 p2p channels, 2 p2p channels per peer
3ccc2fa29ce8:192:225 [1] NCCL INFO comm 0x7f1524003010 rank 1 nranks 4 cudaDev 1 busId 200000 - Init COMPLETE
3ccc2fa29ce8:192:226 [2] NCCL INFO comm 0x7f1518003010 rank 2 nranks 4 cudaDev 2 busId 300000 - Init COMPLETE
3ccc2fa29ce8:192:224 [0] NCCL INFO comm 0x7f1520003010 rank 0 nranks 4 cudaDev 0 busId 100000 - Init COMPLETE
3ccc2fa29ce8:192:227 [3] NCCL INFO comm 0x7f1510003010 rank 3 nranks 4 cudaDev 3 busId 400000 - Init COMPLETE
3ccc2fa29ce8:192:192 [0] NCCL INFO Launch mode Parallel
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/cuda/nccl.py", line 104, in all_gather
    torch._C._nccl_all_gather(inputs, outputs, streams, comms)
RuntimeError: false INTERNAL ASSERT FAILED at "../torch/csrc/cuda/nccl.cpp":227, please report a bug to PyTorch. NCCL COMM NONBLOCKING USED WITH UNSUPPORTED NCCL VERSION.
>>>
```

### Versions

Torch 2.1.0 (Nightly)
NCCL 2.12.12
Python 3.8
CUDA 11.7

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu

**Comment by eqy** [comment]
Should be fixed in #99679.

**Comment by ajindal1** [comment]
Yes, fixed now. Probably by #99679.
